common/lib/xmodule/xmodule/modulestore/tests/test_mongo_call_count_import_export.py

```diff
@@ -0,0 +1,119 @@
+when using the Split modulestore.
+"""
+
+from tempfile import mkdtemp
+from shutil import rmtree
+from unittest import TestCase
+
+from xmodule.modulestore.xml_importer import import_from_xml
+from xmodule.modulestore.xml_exporter import export_to_xml
+from xmodule.modulestore.tests.factories import check_mongo_calls
+from xmodule.modulestore.tests.test_cross_modulestore_import_export import (
+    MongoContentstoreBuilder, MixedModulestoreBuilder, VersioningModulestoreBuilder,
+    TEST_DATA_DIR
+)
+
+MIXED_SPLIT_MODULESTORE_BUILDER = MixedModulestoreBuilder([('split', VersioningModulestoreBuilder())])
+
+
+class CountMongoCallsXMLRoundtrip(TestCase):
+    """
+    This class exists to test XML import and export to/from Split.
+    """
+
+    def setUp(self):
+        super(CountMongoCallsXMLRoundtrip, self).setUp()
+        self.export_dir = mkdtemp()
+        self.addCleanup(rmtree, self.export_dir, ignore_errors=True)
+
+    def test_import_export(self):
+        # Construct the contentstore for storing the first import
+        with MongoContentstoreBuilder().build() as source_content:
+            # Construct the modulestore for storing the first import (using the previously created contentstore)
+            with MIXED_SPLIT_MODULESTORE_BUILDER.build(source_content) as source_store:
+                # Construct the contentstore for storing the second import
+                with MongoContentstoreBuilder().build() as dest_content:
+                    # Construct the modulestore for storing the second import (using the second contentstore)
+                    with MIXED_SPLIT_MODULESTORE_BUILDER.build(dest_content) as dest_store:
+                        source_course_key = source_store.make_course_key('a', 'course', 'course')
+                        dest_course_key = dest_store.make_course_key('a', 'course', 'course')
+
+                        with check_mongo_calls(16, 190):
+                            import_from_xml(
+                                source_store,
+                                'test_user',
+                                TEST_DATA_DIR,
+                                course_dirs=['manual-testing-complete'],
+                                static_content_store=source_content,
+                                target_course_id=source_course_key,
+                                create_course_if_not_present=True,
+                                raise_on_failure=True,
+                            )
+
+                        with check_mongo_calls(37):
+                            export_to_xml(
+                                source_store,
+                                source_content,
+                                source_course_key,
+                                self.export_dir,
+                                'exported_source_course',
+                            )
+
+                        with check_mongo_calls(16, 189):
```

Comment:
The one less call in the second import stems from the wiki_slug. When the
wiki_slug is imported from the test course (manual-testing-complete)
initially, the course id in the XML doesn't match the course id into which
the course XML is imported. So an extra call is made to update the
modulestore after the mismatch is discovered.
In the second import, the wiki_slug and the course_id already matches - and
an extra update is not required.

On Tue, Feb 10, 2015 at 9:18 PM, Nimisha Asthagiri <notifications@github.com
> wrote:

> In
> common/lib/xmodule/xmodule/modulestore/tests/test_mongo_call_count_import_export.py
> <https://github.com/edx/edx-platform/pull/6920#discussion_r24470551>:
>
> > +                                static_content_store=source_content,
> > +                                target_course_id=source_course_key,
> > +                                create_course_if_not_present=True,
> > +                                raise_on_failure=True,
> > +                            )
> > +
> > +                        with check_mongo_calls(37):
> > +                            export_to_xml(
> > +                                source_store,
> > +                                source_content,
> > +                                source_course_key,
> > +                                self.export_dir,
> > +                                'exported_source_course',
> > +                            )
> > +
> > +                        with check_mongo_calls(16, 189):
>
> Just curious, why is this 1 less than the previous call to import?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/6920/files#r24470551>.
>